### DISCUSSION TITLE
RevoGrid Events and some unclear behaviour or usage

### DISCUSSION THREAD

**Discussion by Thoemmy** [revolist/revogrid#479] [General]
First at all, i really like your library and the component you are providing, thats why i am really motivated to work with it and to make it better. I am new to revogrid as you know i guess, so some things are unclear to me and thats why i started this discussion.

I was wondering about the existing events and the data they provide on change e.g., so i've tested a couple of these events and some questions came up which i want to ask here.

### Events:

**beforeedit**
description: This event is triggered before the data is edited.
Issue: Usecase is unclear cause event gots fired a couple miliseconds before afterEdit and event doesnt carry the value before it gots edited in it. It has the same value like afteredit

<img width="1040" alt="Bildschirmfoto 2024-07-21 um 17 49 43" src="https://github.com/user-attachments/assets/45c61eb7-6eea-4f76-9ef6-c53af4c83831">

---

**beforeeditstart**
description: Emitted before editing starts.
Issue: the property "val" in the object detail is empty, the only way to get the initial val of the cell is through the model object, is that intended to be like that?

<img width="1140" alt="Bildschirmfoto 2024-07-21 um 17 52 15" src="https://github.com/user-attachments/assets/f1573910-669f-430f-8c86-6f3cc511a699">
P.S. If i open the detail object (of beforeeditstart) in the console immediatly after it gots logged then the original value is inside the detail object, but if i do not touch it initially and open it later (after the change), then the original value is gone and only the changed value is shown in the object. As you can see in the screenshot that beforeeditstart has the same value like the beforeedit and afteredit, maybe i do something wrong??

---

**beforerange**
description: Triggered before range applied.
Issue: See screenshots

I range edited the column Country which should be column index 3. I copied the value "Lee County" to the next three rows.

<img width="597" alt="Bildschirmfoto 2024-07-21 um 17 36 55" src="https://github.com/user-attachments/assets/94ba60e8-8140-4c77-8d9b-604e79688389">

The event gave me following output:

<img width="885" alt="Bildschirmfoto 2024-07-21 um 17 37 11" src="https://github.com/user-attachments/assets/ec1b7d83-5da2-46ad-94b0-5ed754b920d7">

My question is, why is the index not working or do i missunderstood it?
Cause we can see on the screenshot that the columnindex is 2 and the rowindex is 0.
Additionally should the previous data also be carried out by this event?

---

I hope i do not annoy you with my questions ^^, if so please don't let me know -> emotional damage, just kidding ;)
If i asked something stupid then you can tell me directly or if you need some further examples or information, don't hesitate to ask me for it.

i am looking forward to hear from you soon.

**Comment by revolist** [comment]
Hello @Thoemmy,

Thank you for being an active part of the community. We welcome all questions, though we may not always have the time to answer each one promptly. However, we do our best to respond as quickly as possible.

Regarding your first question about the event descriptions, could you please summarize your query for us? It seems that you might be unclear about the purpose of these events. Understanding your specific goals might help us provide a more direct solution.

For your last question, it appears you might be referring to detail.mapping. If this is the case, the rowIndex: 0 property is used because it maps data starting from row 0. This is an advanced property for low-level customization.

Please summarize your questions and let us know what you find lacking in the documentation. We will strive to update it for future reference. You are also welcome to make updates yourself and submit a pull request.

Thank you.

**Reply by Thoemmy** [reply]
Thank you for the response, i will work on a detailed example and also provide information about my goals.
This might take up some time but i will respond as soon as possible.
